sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -279,14 +282,26 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
           overwrite = false,
           ifPartitionNotExists)
       case table: InsertOverwriteTableContext =>
-        val (relation, cols, partition, ifPartitionNotExists) = visitInsertOverwriteTable(table)
+        val (relation, cols, partition, ifPartitionNotExists, _)
+            = visitInsertOverwriteTable(table)
         InsertIntoStatement(
           relation,
           partition,
           cols,
           query,
           overwrite = true,
           ifPartitionNotExists)
+      case table: InsertIntoReplaceWhereContext =>
+        val (relation, cols, partition, ifPartitionNotExists, replacePredicates)
+            = visitInsertIntoReplaceWhere(table)
+        InsertIntoStatement(
```

Comment:
can't we return `OverwriteByExpression` directly here? Then we don't need to touch `InsertIntoStatement`. This is v2 code path only, and we don't need to use `InsertIntoStatement` and covert it to either v1 or v2 code path later.